auth-server/src/utils/jwtHelpers.js

```diff
@@ -1,82 +1,46 @@
 const jwt = require("jsonwebtoken");
 
-function signAuthJWT(id, password, role) {
-  if (!password || !id || !role) {
-    throw "Cannot create hash without id, password, and role!";
+function signAuthJWT(email, role) {
+  if (!email || !role) {
+    throw "Cannot create hash without email and role!";
   }
-  return jwt.sign(
-    { userId: id, hashedPassword: password, permission: role },
-    process.env.AUTH_SECRET,
-    {
-      expiresIn: "1d"
-    }
-  );
+  return jwt.sign({ sub: email, permission: role }, process.env.AUTH_SECRET, {
+    expiresIn: "1d"
+  });
 }
 
 // Return true if the JWT is valid and matches the parameters
-function verifyAuthJWT(token, id, password) {
+function verifyAuthJWT(token, userEmail) {
   try {
-    let { userId, hashedPassword } = jwt.verify(token, process.env.AUTH_SECRET);
-    if (String(userId) === String(id) && hashedPassword == password) {
+    let { email } = jwt.verify(token, process.env.AUTH_SECRET);
```

Comment:
What happens if the user passes a token that does not contain an email field. Does jwt.verify return a proper error that will be handled by line 16?

Reply:
hmm i'm not too sure about that, i can look into this later!

Reply:
hm ok i looked at the code for jwt.verity(), i didn't see anything that would return the proper error so i can just check manually